lib/scala/cli/src/main/scala/org/enso/cli/task/TaskProgress.scala

```diff
@@ -24,22 +17,14 @@ trait TaskProgress[A] {
     */
   def addProgressListener(listener: ProgressListener[A]): Unit
 
-  /** Blocks and waits for the completion of the task.
-    *
-    * Optionally displays a progress bar in the terminal. Returns a [[Try]]
-    * value that wraps the result.
-    *
-    * @param showProgress whether or not to show a progress bar while waiting
-    */
-  def waitForResult(showProgress: Boolean = false): Try[A] =
-    if (showProgress) ProgressBar.waitWithProgress(this)
-    else TaskProgress.waitForTask(this)
-
   /** Waits for the task to finish and returns its value, throwing any
     * exceptions that were reported.
     */
   def force(): A = TaskProgress.waitForTask(this).get
```

Comment:
Well, what we had earlier was `waitForResult(...).get` which is equivalent to `force`, now it is just more apparent.

The runtime version manager and the whole launcher sometimes use `Try` to manage exceptions but most of the top-level functions just throw. In the launcher that is caught in the top-level to display the error and terminate the process. In the project manager service it is handled by being wrapped in `blockingOp` and handling these possible exceptions.

I decided to go for throwing in these places, as besides things that I do throw, there are exceptions like `FileNotExists` etc. that can happen there and I was afraid I may forget to wrap them in a proper `Try` or other constructs. I guess it may not the best design and maybe I should have wrapped all of these into `Either` or at least `Try` but I thought that they offer little advantage in the context that these methods were being written.

Especially as I needed quite a lot of local error handling for rollbacks etc. and `try ... catch ... finally` was simpler to write than using a lot of `recoverWith` on `Try` (is there even a `finally` counterpart?).

I think it was due to the fact that a lot of launcher's logic, like installing stuff is highly imperative (as it involves doing a lot of sequential operations and handling their failures). Do you think it would have be better to use `Try` there? I guess I can't rewrite all of this now (as this is big parts of the version management), but it may be a guide in which direction I should move stuff when refactoring/updating various components.